### PR TITLE
Add Code Snippet CLI documentation and refactor existing content

### LINKED ISSUE #1502: Code snippet documentation does not include CLI usage information
**Is your feature request related to a problem? Please describe.**
The code snippet documentation at https://elyra.readthedocs.io/en/latest/user_guide/code-snippets.html does not cover how code snippets can be managed using the CLI.

**Describe the solution you'd like**
 See https://elyra.readthedocs.io/en/latest/user_guide/runtime-conf.html#managing-runtime-configurations-using-the-elyra-cli and https://elyra.readthedocs.io/en/latest/user_guide/runtime-image-conf.html#managing-runtime-images-with-the-command-line-interface

### PR DESCRIPTION
**Pull request by ptitzler**
This PR
 - updates the existing code snippet user guide content
 - adds a link from the Elyra CLI topic in the user guide to the new content
 - closes #1502 
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: docs/source/user_guide/code-snippets.md [hunk 1]
```diff
@@ -21,62 +21,151 @@ The ability to reuse pieces of code allows users to avoid doing repetitive work,
...
-			"   return num % 2 == 0"
-		],
-        "tags": [
-		  "python"
-        ]
-	}
-}
+Schema   	Instance		Resource  
+---------   	--------		--------
+code-snippet   	preview_dataframe	/.../metadata/preview_dataframe.json
```
Comment (karlaspuldaro) [inline docs/source/user_guide/code-snippets.md:88]: shouldn't the path be `/.../metadata/code-snippets/preview_dataframe.json ` ?

File: docs/source/user_guide/code-snippets.md [hunk 2]
```diff
@@ -21,62 +21,151 @@ The ability to reuse pieces of code allows users to avoid doing repetitive work,
...
+
+Example: `Preview DataFrame`
+
+##### N/A (name)
+
+A unique identifier for this code snippet. If not specified when the code snippet is created, a value is automatically generated from `display_name`.
+
+Example: `preview_dataframe`
+
+##### description (description)
```
Comment (karlaspuldaro) [inline docs/source/user_guide/code-snippets.md:149]:
```suggestion
##### Description (description)
```

File: docs/source/user_guide/code-snippets.md (hunk 1 above)
Comment (ptitzler) [reply docs/source/user_guide/code-snippets.md:88]: jap

File: docs/source/user_guide/code-snippets.md [hunk 3]
```diff
@@ -21,62 +21,151 @@ The ability to reuse pieces of code allows users to avoid doing repetitive work,
...
+```
+
+Refer to the [Code snippet properties](#code-snippet-properties) section for an explanation of the parameters. Note that you must specify the `--name` parameter and the `--replace` parameter. 
+
+#### Deleting a code snippet
+
+To delete a code snippet run the following command, replacing the code snippet name as appropriate.
+
+```bash
+elyra-metadata remove code-snippet --name="preview_dataframe"
```
Comment (karlaspuldaro) [inline docs/source/user_guide/code-snippets.md:130]: This command throws the error `Subcommand 'code-snippet' is invalid.`
```suggestion
elyra-metadata remove code-snippets --name="preview_dataframe"
```

Comment (ptitzler) [reply docs/source/user_guide/code-snippets.md]: arg, yes. thank you. I never understood why we use plural, even though most commands operate on a single instance.

**Review by karlaspuldaro — APPROVED** [review]
LGTM :)

**Review by lresende — APPROVED** [review]

**Pull request merged**